lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -387,6 +393,20 @@ def configure(conf)
                   ' ignored.'
       end
 
+      # 1. If @metadata_agent_url is customized (aka not nil), use that.
+      # 2. Otherwise check the presence of the environment variable
+      #    STACKDRIVER_METADATA_AGENT_URL and use that if set.
+      # 3. Fall back to the default if neither is set.
+      if @enable_metadata_agent
+        # Convert to string to capture empty string.
+        @metadata_agent_url ||=
+          if ENV[METADATA_AGENT_URL_ENV_VAR_NAME].to_s.empty?
```

Comment:
Just curious, could this also be:
```ruby
@metadata_agent_url ||= ENV[METADATA_AGENT_URL_ENV_VAR_NAME]
@metadata_agent_url ||= DEFAULT_METADATA_AGENT_URL
```
? Or would it miss some cases?

Reply:
This way it will miss the empty string case. If `ENV[METADATA_AGENT_URL_ENV_VAR_NAME]` is an empty string, `@metadata_agent_url` will end up being an empty string.

Reply:
Got it. I thought we didn't really need the empty string case, but then realized that it allows one to invoke the agent as `STACKDRIVER_METADATA_AGENT_URL= metadatad` (although it's also possible to unset the variable completely via `/usr/bin/env -u STACKDRIVER_METADATA_AGENT_URL metadatad`). I'm ok with leaving it.